lib/ansible/modules/cloud/pubnub/pubnub_blocks.py

```diff
@@ -0,0 +1,2090 @@
+  - Sergey Mamontov <sergey@pubnub.com> (@parfeon)
+requirements:
+  - "python >= 2.7"
+options:
+  email:
+    description:
+      - Email from account for which new session should be started.
+      - Not required if C(cache) contains result of previous module call (in same play).
+    required: false
+  pwd:
+    description:
+      - Password which match to account to which specified C(email) belong.
+      - Not required if C(cache) contains result of previous module call (in same play).
+    required: false
+  cache:
+    description:
+      - In case if single play use blocks management module few times it is preferred to enabled 'caching' by making previous module to share gathered artifacts and pass them to this parameter.
+    required: false
+    default: {}
+  application:
+    description:
+      - Name of target PubNub application for which blocks configuration on specific C(keyset) will be done.
+    required: true
+  keyset:
+    description:
+      - Name of application's keys set which is bound to managed blocks.
+    required: true
+  state:
+    description:
+      - Intended block state after event handlers creation / update process will be completed.
+    required: false
+    default: 'start'
+    choices: ['start', 'stop', 'present', 'absent']
+  block:
+    description:
+      - Name of managed block which will be later visible on admin.pubnub.com.
+    required: true
+  description:
+    description:
+      - Short block description which will be later visible on admin.pubnub.com. Used only if block doesn't exists and won't change description for existing block.
+    required: false
+    default: 'New block'
+  event_handlers:
+    description:
+      - List of event handlers which should be updated for specified C(block).
+      - "Each entry for new event handler should contain: C(name), C(src), C(channels), C(event). C(name) used as event handler name which can be used later to make changes to it."
+      - C(src) is full path to file with event handler code.
+      - C(channels) is name of channel from which event handler is waiting for events.
+      - "C(event) is type of event which is able to trigger event handler: I(js-before-publish), I(js-after-publish), I(js-after-presence)."
+      - Each entry for existing handlers should contain C(name) (so target handler can be identified). Rest parameters (C(src), C(channels) and C(event)) can be added if changes required for them.
+      - It is possible to rename event handler by adding C(changes) key to event handler payload and pass dictionary, which will contain single key C(name), where new name should be passed.
+      - To remove particular event handler it is possible to set C(state) for it to C(absent) and it will be removed.
+    required: false
+    default: []
+  changes:
+    description:
+      - List of fields which should be changed by block itself (doesn't affect any event handlers).
+      - Possible options for change is C(name) and C(description).
```

Comment:
@ryansb I forgot to update this line in docs. Actually it allow to change only name of block (if `changes` dictionary appear in event handlers with name key it will rename event handler as well).  
Because module use `name` of block and it's event handlers to manipulate, it is impossible to specify new name in `name` field of module call. `new_name` - this kind of key doesn't look good as for me, that is why I suggested to add expected `changes` in block separately.